tests/migrations/test_base.py

```diff
@@ -225,6 +227,11 @@ def set_up_test_model(
                 'Pony',
                 models.Index(fields=['pink', 'weight'], name='pony_test_idx'),
             ))
+        if expression_index:
+            operations.append(migrations.AddIndex(
+                'Pony',
+                models.Index(Abs('weight'), name='pony_test_fun_idx'),
```

Comment:
Can we add also a test for `F()` with transforms, e.g.
```python
models.Index(F('weight__abs'), name='pony_test_fun_idx'),
```

Reply:
Hmm not sure if that's possible or if I'm missing something:
```python
Index(Lower('name'), F('name__upper'), name='lower_upper_func_idx')
```
```
======================================================================
ERROR [0.030s]: test_composite_func_index (schema.tests.SchemaTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tests/django/django/db/models/sql/query.py", line 1641, in transform
    return self.try_transform(wrapped, name)
  File "/tests/django/django/db/models/sql/query.py", line 1225, in try_transform
    "permitted%s" % (name, output_field.__name__, suggestion)
django.core.exceptions.FieldError: Unsupported lookup 'upper' for CharField or join on the field not permitted.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/tests/django/django/test/testcases.py", line 1314, in skip_wrapper
    return test_func(*args, **kwargs)
  File "/tests/django/tests/schema/tests.py", line 2529, in test_composite_func_index
    editor.add_index(Author, index)
  File "/tests/django/django/db/backends/postgresql/schema.py", line 208, in add_index
    self.execute(index.create_sql(model, self, concurrently=concurrently), params=None)
  File "/tests/django/django/db/models/indexes.py", line 111, in create_sql
    include=include, expressions=self.expressions, **kwargs,
  File "/tests/django/django/db/backends/postgresql/schema.py", line 226, in _create_index_sql
    include=include, expressions=expressions
  File "/tests/django/django/db/backends/base/schema.py", line 1033, in _create_index_sql
    table, expressions, compiler, self.quote_value, opclasses
  File "/tests/django/django/db/backends/ddl_references.py", line 211, in __init__
    expression.resolve_expression(self.compiler.query) for expression in expressions
  File "/tests/django/django/db/backends/ddl_references.py", line 211, in <listcomp>
    expression.resolve_expression(self.compiler.query) for expression in expressions
  File "/tests/django/django/db/models/expressions.py", line 580, in resolve_expression
    return query.resolve_ref(self.name, allow_joins, reuse, summarize)
  File "/tests/django/django/db/models/sql/query.py", line 1757, in resolve_ref
    transform = join_info.transform_function(targets[0], final_alias)
  File "/tests/django/django/db/models/sql/query.py", line 1645, in transform
    raise last_field_exception
  File "/tests/django/django/db/models/sql/query.py", line 1623, in setup_joins
    names[:pivot], opts, allow_many, fail_on_missing=True,
  File "/tests/django/django/db/models/sql/query.py", line 1570, in names_to_path
    " not permitted." % (names[pos + 1], name))
django.core.exceptions.FieldError: Cannot resolve keyword 'upper' into field. Join on 'name' not permitted.
```

Reply:
You need to register it:
```python
with register_lookup(CharField, Upper), register_lookup(CharField, Lower):
    Index(F('name__lower'), F('name__upper'), name='lower_upper_func_idx')
```

Reply:
I added the test `tests.schema.tests.SchemaTests.test_func_index_with_lookups`